### PR TITLE
Improvements for developers builds & fixes

### PR DESCRIPTION
**Pull request by andr2000**
These are improvements for developers builds, e.g. not using build system scripts.
It allows:
1. making inner builds use sstate cache mirrors
2. correctly set build history location

Fix problem with machine name clashes

### CODE REVIEW

File: meta/conf/machine/salvator-x-m3.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+require xt-distro-machine.inc
+
+MACHINEOVERRIDES =. "rcar:rcar-gen3:"
```
Comment (aanisov) [inline meta/conf/machine/salvator-x-m3.conf]: I'm just curious why did you miss r8a7796 SoC name here? Did you just discard it as meaningless?

Comment (andr2000) [inline meta/conf/machine/salvator-x-m3.conf]: this is for outer yocto, no need for SOC at this level

File: meta/classes/build_yocto.bbclass [hunk 2]
```diff
@@ -97,7 +112,7 @@ do_collect_build_history() {
 do_populate_sstate_cache() {
     if [ -n ${XT_SSTATE_CACHE_MIRROR_DIR} ] ; then
         install -d ${XT_SSTATE_CACHE_MIRROR_DIR}
-        cp -rf ${SSTATE_DIR}/${PN}/* ${XT_SSTATE_CACHE_MIRROR_DIR}
+        cp -rf ${SSTATE_DIR}/${PN}/* ${XT_SSTATE_CACHE_MIRROR_DIR} || true
```
Comment (lorc) [inline meta/classes/build_yocto.bbclass]: have you tried `cp -d` ? and '-u`, by the way?

File: meta/conf/machine/salvator-x-m3.conf (hunk 1 above)
Comment (lorc) [inline meta/conf/machine/salvator-x-m3.conf]: Your commit message states that this change allows us to distinguish between, say, M3 and H3. I can't see how it is done by this change.

File: meta/classes/base.bbclass [hunk 3]
```diff
@@ -84,7 +84,7 @@ base_update_conf_value() {
         echo "${key}${key_prefix}=" >> "${config_file}"
     fi
     # substitute
-    sed -i "s%\(^${key} *[?:]*= *\).*%\1\"${value}\"%" "${config_file}"
+    sed -i "s%\(^${key} *[?+:]*= *\).*%\1\"${value}\"%" "${config_file}"
```
Comment (lorc) [inline meta/classes/base.bbclass]: Why you didn't squashed this change with https://github.com/xen-troops/xt-distro/commit/a4d4fd577f2101a1c37360b02c6576ce65f6f66a?

File: meta/conf/machine/salvator-x-m3.conf (hunk 1 above)
Comment (andr2000) [inline meta/conf/machine/salvator-x-m3.conf]: instead of having a single Salvator-X machine we now have salvator-x-h3 and salvator-x-m3 machines, thus we can tell what kind of SOM is installed into salvator-x board

File: meta/classes/build_yocto.bbclass (hunk 2 above)
Comment (andr2000) [inline meta/classes/build_yocto.bbclass]: yes, I did

File: meta/classes/base.bbclass (hunk 3 above)
Comment (andr2000) [inline meta/classes/base.bbclass]: this adds/fixes new functionality, e.g. "+" is allowed now, so I decided it deserves its own patch

Comment (andr2000) [inline meta/classes/base.bbclass]: I'll squash

**Comment by lorc** [comment]
After https://github.com/xen-troops/xt-distro/pull/2/commits/c657075044ed1ba59b521be4485662f882de2302 will be squashed into https://github.com/xen-troops/xt-distro/pull/2/commits/a4d4fd577f2101a1c37360b02c6576ce65f6f66a:

`Reviewed-by: Volodymyr Babchuk <vlad.babchuk@gmail.com>`
